pkg/runtime/runtime.go

```diff
@@ -1424,7 +1424,7 @@ func (a *DaprRuntime) getGRPCAPI() grpc.API {
 }
 
 func (a *DaprRuntime) getPublishAdapter() runtimePubsub.Adapter {
-	if len(a.pubSubs) == 0 {
+	if len(a.pubSubs) == 0 && !a.runtimeConfig.Standalone.EnableDynamicLoading {
```

Comment:
Why is this condition added here?

Reply:
It is to pass the pubsubAdapter to the api even if currently there are no pubsub components in the runtime. Since the API servers get initialised during dapr runtime init, incase we want to add pubsub components later through dynamic loading and use it using API, we need the adapter. Tweaked the condition and added a comment for more clarity.